File: Composicion y Agregacion/Ejercicio7/Python/Estudiante.py
class Estudiante:
    def __init__(self, nombre, carrera, semestre):
        self.nombre = nombre
        self.carrera = carrera
        self.semestre = semestre
    def mostrar(self):
        print("Nombre:", self.nombre)
        print("Carrera:", self.carrera)
        print("Semestre:", self.semestre)
class Universidad:
    def __init__(self, nombre):
        self.nombre = nombre
        self.estudiantes = []
    def agregar(self, estudiante):
        self.estudiantes.append(estudiante)
    def mostrar(self):
        print("Universidad:", self.nombre)
        for e in self.estudiantes:
            e.mostrar()
u = Universidad("UMSA")
u.agregar(Estudiante("Ana", "Medicina", 2))
u.agregar(Estudiante("Luis", "Derecho", 4))
u.mostrar()

File: Composicion y Agregacion/Ejercicio7/Java/Estudiante.java
import java.util.ArrayList;
class Estudíante {
    public String nombre;
    public String carrera;
    public int semestre;
    public Estudíante(String nombre, String carrera, int semestre) {
        this.nombre = nombre;
        this.carrera = carrera;
        this.semestre = semestre;
    }
    public void mostrar() {
        System.out.println("Estudíante: " + nombre + ", Carrera: " + carrera + ", Semestre: " + semestre);
    }
}
class Universidad {
    public String nombre;
    public ArrayList<Estudíante> estudiantes;
    public Universidad(String nombre) {
        this.nombre = nombre;
        this.estudiantes = new ArrayList<>();
    }
    public void agregar(Estudíante e) {
        estudiantes.add(e);
    }
    public void mostrar() {
        System.out.println("Universidad: " + nombre);
        for (Estudíante e : estudiantes) {
            e.mostrar();
        }
    }
}
public class Estudiante {
    public static void main(String[] args) {
        Universidad u = new Universidad("Univalle");
        Estudíante e1 = new Estudíante("Ana", "Ingeniería", 3);
        Estudíante e2 = new Estudíante("Luis", "Medicina", 5);
        u.agregar(e1);
        u.agregar(e2);
        u.mostrar();
    }
}
